frontend/app/routes/protected/multi-channel/edit-application/validation.server.ts

```diff
@@ -0,0 +1,114 @@
+import * as v from 'valibot';
+
+import { getSinCaseService } from '~/.server/domain/multi-channel/case-api-service';
+import type { SinCaseDto } from '~/.server/domain/multi-channel/sin-case-models';
+import { i18nRedirect } from '~/.server/utils/route-utils';
+import { HttpStatusCodes } from '~/errors/http-status-codes';
+import { primaryDocument } from '~/routes/protected/sin-application/validation.server';
+import { toISODateString } from '~/utils/date-utils';
+
+export async function fetchSinCase(session: AppSession, caseId: string) {
+  if (session.editingCase && session.editingCase.caseId === caseId) {
+    return session.editingCase;
+  }
```

Comment:
Storing a chunk of data like this in session is not multi-tab safe. If the user opens the same flow in multiple tabs, the data from one will clobber the data from the other.

Does the data *need* to be stored in session? Is this a part of a multi-page flow? If not, just fetch the data on demand and updated it as required. If this is a part of a multi-page flow and you need to keep the data from one page to the next (and you can't re-fetch it), then you have to use a tab id to isolate the session data.